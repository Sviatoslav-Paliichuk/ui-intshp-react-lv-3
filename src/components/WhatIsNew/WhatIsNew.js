import React from 'react';

import ItemInfo from './ItemInfo';
import Heading from '../Heading';

import './WhatIsNew.scss';

const CN = 'what-is-new';

const WhatIsNew = (props) => {
  const { items } = props;
  const itemsToRender = items.slice(0, 3).sort(
    (a, b) => new Date(b.createdAt) - new Date(a.createdAt)
  );
  const block = itemsToRender.map((item) => (
    <ItemInfo className={`${CN}-wrapper__item`} key={item._id} item={item} />
  ));

  return (
    <div className={`content ${CN}`}>
      <Heading className={`${CN}__title}`} title="What is new?" position="center" />
      <div className={`container ${CN}-wrapper`}>
        {block}
      </div>
    </div>
  );
};


export default WhatIsNew;
