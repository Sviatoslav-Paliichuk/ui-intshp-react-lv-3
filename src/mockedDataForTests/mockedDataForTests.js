export const mockedBrands = [
  {
    id: 'Boss',
    photo: 'https://i.pinimg.com/originals/a6/c8/7b/a6c87bc1494906db8c9bafb3a4f4686b.png'
  },
  {
    id: 'DonnaKaran',
    photo: 'https://i.pinimg.com/originals/86/5a/64/865a64b6060a7f40b94301a7fab8bbcf.jpg'
  },
  {
    id: 'Gucci',
    photo: 'https://myrealdomain.com/images/gucci-logo-png-1.png'
  },
  {
    id: 'Baldinini',
    photo: 'https://upload.wikimedia.org/wikipedia/en/thumb/f/f9/Baldinini_logo.svg/1280px-Baldinini_logo.svg.png'
  },
  {
    id: 'Dior',
    photo: 'https://pervayaoptica.ru//upload//DIOR.png'
  },
  {
    id: 'Gucci2',
    photo: 'https://myrealdomain.com/images/gucci-logo-png-1.png'
  }
];

export const mockedProductList = [
  {
    id: '1',
    image: 'https://img.ans-media.com/files/sc_staging_images/product/full_img_2033080.jpg',
    title: 'Awesome dress',
    collection: 'spring collection 2016',
    category: 'women',
    label: 'sale',
    sizes: '[S, M, L]',
    price: '199'
  },
  {
    id: '2',
    image: 'https://img.ans-media.com/files/sc_staging_images/product/large_img_2033076.jpg',
    title: 'Beautiful dress',
    collection: 'spring collection 2016',
    category: 'women',
    label: 'new',
    sizes: '[S, M, L]',
    price: '299'
  },
  {
    id: '3',
    image: 'https://img.ans-media.com/files/sc_staging_images/product/large_img_2033787.jpg',
    title: 'Amazing dress',
    collection: 'spring collection 2016',
    category: 'women',
    label: '',
    sizes: '[M, L]',
    price: '399'
  }
];

export const mockedBlogsPage = [
  {
    id: 1,
    date: '12 october 2019',
    title: 'summer trends',
    intro: 'we know what you want this summer. most fashion looks',
    photo: '1.jpg',
    labels: ['summer', 'swimsuit', 'bags', 'sunglasses'],
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc,'
  },
  {
    id: 2,
    date: '11 january 2018',
    title: 'Free-spirited style for boho-chic',
    intro: 'new trends comming through',
    photo: '2.jpg',
    labels: ['boho', 'summer', 'trend', 'style'],
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc,'
  },
  {
    id: 3,
    date: '25 may 2019',
    title: 'military style',
    intro: 'military style boots',
    photo: '3.jpg',
    labels: ['boots', 'autumn', 'trend', 'style', 'military'],
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc,'
  }
];

export const mockedItems = [
  {
    _id: '5db069a4827ca80017ca5a28',
    sizes: ['S', 'M', 'L', 'XL', 'XXL'],
    colors: ['blue'],
    genders: ['man'],
    seasons: ['spring', 'summer', 'autumn', 'winter'],
    imageIds: [],
    collectionIds: [],
    quantity: 10,
    sellCount: 0,
    price: 499,
    title: 'Jeans shirt',
    description: 'The classic nature of the ultimate must-have item, returns intent on representing your style to its fullest. Shirt made from jeans material, with a small collar, yoke on the back and rounded cuffs.',
    brandId: '5daf191ba3f2fe001790b0a0',
    category: 'shirts',
    createdAt: '2019-10-23T14:54:28.157Z',
    updatedAt: '2019-10-23T14:54:28.157Z',
    __v: 0,
    images: [
      {
        _id: '5db1790434a29600172f6e85',
        productId: '5db069a4827ca80017ca5a28',
        claudinaryId: 'armani-jeansblue-1.1',
        productColor: 'blue',
        __v: 0
      },
      {
        _id: '5db1790e34a29600172f6e86',
        productId: '5db069a4827ca80017ca5a28',
        claudinaryId: 'armani-jeansblue-1.2',
        productColor: 'blue',
        __v: 0
      },
      {
        _id: '5db1791d34a29600172f6e87',
        productId: '5db069a4827ca80017ca5a28',
        claudinaryId: 'armani-jeansblue-1.3',
        productColor: 'blue',
        __v: 0
      },
      {
        _id: '5db1792534a29600172f6e88',
        productId: '5db069a4827ca80017ca5a28',
        claudinaryId: 'armani-jeansblue-1.4',
        productColor: 'blue',
        __v: 0
      }
    ],
    brand: [
      {
        _id: '5daf191ba3f2fe001790b0a0',
        title: 'Emporio Armani',
        logo: 'fnllqjn238hr9u8h21n21',
        __v: 0
      }
    ]
  },
  {
    _id: '5db179b634a29600172f6e8a',
    sizes: ['S', 'M'],
    colors: ['black'],
    genders: ['woman'],
    seasons: ['spring, summer, autumn'],
    collectionIds: [],
    quantity: 10,
    sellCount: 0,
    price: 200,
    title: 'Black shirt with white stripes',
    description: 'Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.',
    brandId: '5daf1650a3f2fe001790b09b',
    category: 'shirts',
    createdAt: '2019-10-24T10:15:18.279Z',
    updatedAt: '2019-10-24T10:15:18.279Z',
    __v: 0,
    images: [{
      _id: '5db17a7d34a29600172f6e8c', productId: '5db179b634a29600172f6e8a', claudinaryId: 'blouse_1_1', productColor: 'black', __v: 0
    }, {
      _id: '5db17a8134a29600172f6e8d', productId: '5db179b634a29600172f6e8a', claudinaryId: 'blouse_1_2', productColor: 'black', __v: 0
    }, {
      _id: '5db17a8434a29600172f6e8e', productId: '5db179b634a29600172f6e8a', claudinaryId: 'blouse_1_3', productColor: 'black', __v: 0
    }, {
      _id: '5db17a8634a29600172f6e8f', productId: '5db179b634a29600172f6e8a', claudinaryId: 'blouse_1_4', productColor: 'black', __v: 0
    }],
    brand: [{
      _id: '5daf1650a3f2fe001790b09b', title: 'Boss', logo: 'cm24icm93cm4rergre', __v: 0
    }]
  },
  {
    _id: '5db17cf434a29600172f6e91',
    sizes: ['S', 'M', 'L'],
    colors: ['white'],
    genders: ['woman'],
    seasons: ['spring', 'summer', 'autumn'],
    collectionIds: [],
    quantity: 8,
    sellCount: 0,
    price: 320,
    title: 'White shirt with black stripes',
    description: 'Soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.',
    brandId: '5daf1650a3f2fe001790b09b',
    category: 'shirts',
    createdAt: '2019-10-24T10:29:08.982Z',
    updatedAt: '2019-10-26T23:00:47.525Z',
    __v: 0,
    images: [{
      _id: '5db17da834a29600172f6e92', productId: '5db17cf434a29600172f6e91', claudinaryId: 'blouse_2_1', productColor: 'white', __v: 0
    }, {
      _id: '5db17daf34a29600172f6e93', productId: '5db17cf434a29600172f6e91', claudinaryId: 'blouse_2_2', productColor: 'white', __v: 0
    }, {
      _id: '5db17db234a29600172f6e94', productId: '5db17cf434a29600172f6e91', claudinaryId: 'blouse_2_3', productColor: 'white', __v: 0
    }, {
      _id: '5db17db434a29600172f6e95', productId: '5db17cf434a29600172f6e91', claudinaryId: 'blouse_2_4', productColor: 'white', __v: 0
    }],
    brand: [{
      _id: '5daf1650a3f2fe001790b09b', title: 'Boss', logo: 'cm24icm93cm4rergre', __v: 0
    }]
  }
];
