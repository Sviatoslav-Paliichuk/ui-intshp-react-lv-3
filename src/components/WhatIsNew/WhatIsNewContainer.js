import { connect } from 'react-redux';

import WhatIsNew from './WhatIsNew';

const mapStateToProps = ({ productsReducer }) => ({
  items: productsReducer.products
});

export default connect(mapStateToProps)(WhatIsNew);
