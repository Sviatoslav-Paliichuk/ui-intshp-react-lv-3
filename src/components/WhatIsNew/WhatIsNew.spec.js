import React from 'react';
import { shallow } from 'enzyme';
import { mockedItems } from '../../mockedDataForTests/mockedDataForTests'

import WhatIsNew from './WhatIsNew';

describe('<WhatIsNew />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WhatIsNew items={mockedItems} />
    );
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
