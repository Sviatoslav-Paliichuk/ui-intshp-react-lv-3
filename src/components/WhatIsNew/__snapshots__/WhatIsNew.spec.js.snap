// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhatIsNew /> should match snapshot 1`] = `
<div
  className="content what-is-new"
>
  <Heading
    className="what-is-new__title}"
    position="center"
    title="What is new?"
  />
  <div
    className="container what-is-new-wrapper"
  >
    <ItemInfo
      className="what-is-new-wrapper__item"
      item={
        Object {
          "__v": 0,
          "_id": "5db17cf434a29600172f6e91",
          "brand": Array [
            Object {
              "__v": 0,
              "_id": "5daf1650a3f2fe001790b09b",
              "logo": "cm24icm93cm4rergre",
              "title": "Boss",
            },
          ],
          "brandId": "5daf1650a3f2fe001790b09b",
          "category": "shirts",
          "collectionIds": Array [],
          "colors": Array [
            "white",
          ],
          "createdAt": "2019-10-24T10:29:08.982Z",
          "description": "Soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
          "genders": Array [
            "woman",
          ],
          "images": Array [
            Object {
              "__v": 0,
              "_id": "5db17da834a29600172f6e92",
              "claudinaryId": "blouse_2_1",
              "productColor": "white",
              "productId": "5db17cf434a29600172f6e91",
            },
            Object {
              "__v": 0,
              "_id": "5db17daf34a29600172f6e93",
              "claudinaryId": "blouse_2_2",
              "productColor": "white",
              "productId": "5db17cf434a29600172f6e91",
            },
            Object {
              "__v": 0,
              "_id": "5db17db234a29600172f6e94",
              "claudinaryId": "blouse_2_3",
              "productColor": "white",
              "productId": "5db17cf434a29600172f6e91",
            },
            Object {
              "__v": 0,
              "_id": "5db17db434a29600172f6e95",
              "claudinaryId": "blouse_2_4",
              "productColor": "white",
              "productId": "5db17cf434a29600172f6e91",
            },
          ],
          "price": 320,
          "quantity": 8,
          "seasons": Array [
            "spring",
            "summer",
            "autumn",
          ],
          "sellCount": 0,
          "sizes": Array [
            "S",
            "M",
            "L",
          ],
          "title": "White shirt with black stripes",
          "updatedAt": "2019-10-26T23:00:47.525Z",
        }
      }
      key="5db17cf434a29600172f6e91"
    />
    <ItemInfo
      className="what-is-new-wrapper__item"
      item={
        Object {
          "__v": 0,
          "_id": "5db179b634a29600172f6e8a",
          "brand": Array [
            Object {
              "__v": 0,
              "_id": "5daf1650a3f2fe001790b09b",
              "logo": "cm24icm93cm4rergre",
              "title": "Boss",
            },
          ],
          "brandId": "5daf1650a3f2fe001790b09b",
          "category": "shirts",
          "collectionIds": Array [],
          "colors": Array [
            "black",
          ],
          "createdAt": "2019-10-24T10:15:18.279Z",
          "description": "Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
          "genders": Array [
            "woman",
          ],
          "images": Array [
            Object {
              "__v": 0,
              "_id": "5db17a7d34a29600172f6e8c",
              "claudinaryId": "blouse_1_1",
              "productColor": "black",
              "productId": "5db179b634a29600172f6e8a",
            },
            Object {
              "__v": 0,
              "_id": "5db17a8134a29600172f6e8d",
              "claudinaryId": "blouse_1_2",
              "productColor": "black",
              "productId": "5db179b634a29600172f6e8a",
            },
            Object {
              "__v": 0,
              "_id": "5db17a8434a29600172f6e8e",
              "claudinaryId": "blouse_1_3",
              "productColor": "black",
              "productId": "5db179b634a29600172f6e8a",
            },
            Object {
              "__v": 0,
              "_id": "5db17a8634a29600172f6e8f",
              "claudinaryId": "blouse_1_4",
              "productColor": "black",
              "productId": "5db179b634a29600172f6e8a",
            },
          ],
          "price": 200,
          "quantity": 10,
          "seasons": Array [
            "spring, summer, autumn",
          ],
          "sellCount": 0,
          "sizes": Array [
            "S",
            "M",
          ],
          "title": "Black shirt with white stripes",
          "updatedAt": "2019-10-24T10:15:18.279Z",
        }
      }
      key="5db179b634a29600172f6e8a"
    />
    <ItemInfo
      className="what-is-new-wrapper__item"
      item={
        Object {
          "__v": 0,
          "_id": "5db069a4827ca80017ca5a28",
          "brand": Array [
            Object {
              "__v": 0,
              "_id": "5daf191ba3f2fe001790b0a0",
              "logo": "fnllqjn238hr9u8h21n21",
              "title": "Emporio Armani",
            },
          ],
          "brandId": "5daf191ba3f2fe001790b0a0",
          "category": "shirts",
          "collectionIds": Array [],
          "colors": Array [
            "blue",
          ],
          "createdAt": "2019-10-23T14:54:28.157Z",
          "description": "The classic nature of the ultimate must-have item, returns intent on representing your style to its fullest. Shirt made from jeans material, with a small collar, yoke on the back and rounded cuffs.",
          "genders": Array [
            "man",
          ],
          "imageIds": Array [],
          "images": Array [
            Object {
              "__v": 0,
              "_id": "5db1790434a29600172f6e85",
              "claudinaryId": "armani-jeansblue-1.1",
              "productColor": "blue",
              "productId": "5db069a4827ca80017ca5a28",
            },
            Object {
              "__v": 0,
              "_id": "5db1790e34a29600172f6e86",
              "claudinaryId": "armani-jeansblue-1.2",
              "productColor": "blue",
              "productId": "5db069a4827ca80017ca5a28",
            },
            Object {
              "__v": 0,
              "_id": "5db1791d34a29600172f6e87",
              "claudinaryId": "armani-jeansblue-1.3",
              "productColor": "blue",
              "productId": "5db069a4827ca80017ca5a28",
            },
            Object {
              "__v": 0,
              "_id": "5db1792534a29600172f6e88",
              "claudinaryId": "armani-jeansblue-1.4",
              "productColor": "blue",
              "productId": "5db069a4827ca80017ca5a28",
            },
          ],
          "price": 499,
          "quantity": 10,
          "seasons": Array [
            "spring",
            "summer",
            "autumn",
            "winter",
          ],
          "sellCount": 0,
          "sizes": Array [
            "S",
            "M",
            "L",
            "XL",
            "XXL",
          ],
          "title": "Jeans shirt",
          "updatedAt": "2019-10-23T14:54:28.157Z",
        }
      }
      key="5db069a4827ca80017ca5a28"
    />
  </div>
</div>
`;
